client/llb/exec.go

```diff
@@ -388,27 +388,26 @@ func (e *ExecOp) Marshal(ctx context.Context, c *Constraints) (digest.Digest, []
 	}
 
 	for _, s := range e.secrets {
-		if s.IsEnv {
+		if s.Env != nil {
 			peo.Secretenv = append(peo.Secretenv, &pb.SecretEnv{
 				ID:       s.ID,
-				Name:     s.Target,
+				Name:     *s.Env,
 				Optional: s.Optional,
 			})
-		} else {
-			pm := &pb.Mount{
-				Input:     pb.Empty,
-				Dest:      s.Target,
-				MountType: pb.MountType_SECRET,
-				SecretOpt: &pb.SecretOpt{
-					ID:       s.ID,
-					Uid:      uint32(s.UID),
-					Gid:      uint32(s.GID),
-					Optional: s.Optional,
-					Mode:     uint32(s.Mode),
-				},
-			}
-			peo.Mounts = append(peo.Mounts, pm)
 		}
+		pm := &pb.Mount{
```

Comment:
This seems to be a behavior change of creating a secret mount when `SecretAsEnv` was set.

If there is a need for mounting the same secret both as file and env (can't think of one) then it should be optional.